### ISSUE TITLE
[BUG] Mismatch in the returned records from similar query with different syntax when used with levenshtein distance %%

### ISSUE THREAD

**Issue by anindyasudhir** [RediSearch/RediSearch#4987]
**Describe the bug**
There is a mismatch the returned records from the following queries: 
Query1: '(@first_name|last_name:(%%Heidi%%))'
Query2: '((@first_name:(%%Heidi%%)) | (@last_name:(%%Heidi%%)))'
According to the documentation, both of these queries, Query1 and Query2, should give the same results but that is not happening when using it with Levenshtein distance (%% ).
 
Query3 '(@first_name|last_name:(Heidi))'
Query4 '((@first_name:(Heidi)) | (@last_name:(Heidi)))'
Query3 and Query4 are giving the same results.

Query5: '((@first_name|last_name:(%%Heidi%%)) (@first_name:"Heidi" @last_name:"Hegi"))' 
Query5 is not returning any records but (@first_name:"Heidi" @last_name:"Hegi") is returning a record.

**To Reproduce**
1.Create hash data in redis with keys, first_name, last_name, address, phone number etc. insert data with names Heidi, Heid, Hegde, etc. Also, insert first_name as Heidi and last_name Hegi to check Query 5.
2. create an index on the hash schema
3. Write search query FT.SEARCH  '(@first_name|last_name:(%%Heidi%%))' and '((@first_name:(%%Heidi%%)) | (@last_name:(%%Heidi%%)))'
4. Compare the results 
5. Also, this query is not returning any   

**Expected behavior**
Both Query1 and query2 should give the same result 

**Environment (please complete the following information):**
 - RediSearch v.2.6.12
- Redis browser workbench

**Comment by nafraf** [comment]
Hi @anindyasudhir, thanks for reaching out!
I'm assuming that you are using the default `DIALECT 1`, and there is an issue that causes that only the first attribute name of the OR operations is read correctly, which can cause the difference.  A possible workaround is to use `DIALECT 2`.

Example:
Data and index creation
```sh
HSET doc1 t1 heidi
HSET doc2 t2 heigi
HSET doc3 t1 hegi
HSET doc4 t1 heidi t2 hegi
FT.CREATE idx schema t1 TEXT t2 TEXT
```
Using `DIALECT 1`, the queries return different results as you reported:
```sh
127.0.0.1:6379> FT.SEARCH idx '(@t1|t2:(%%Heidi%%))' dialect 1 limit 0 0 
1) (integer) 4
127.0.0.1:6379> FT.SEARCH idx '@t1:(%%Heidi%%) | @t2:(%%Heidi%%)' dialect 1 limit 0 0 
1) (integer) 3
```
Using `DIALECT 2`, both queries return the same result:
```sh
127.0.0.1:6379> FT.SEARCH idx '(@t1|t2:(%%Heidi%%))' dialect 2 limit 0 0 
1) (integer) 4
127.0.0.1:6379> FT.SEARCH idx '(@t1|t2:(%%Heidi%%))' dialect 2 
1) (integer) 4
```

**Comment by anindyasudhir** [comment]
Hi @nafraf , 
Thanks for the reply! 
I tested it with Dialect 2 and it still didn't work. Can you check your Dialect 2 queries again, because you have tested Dialect 2 on same queries. 
I tested it on Query5: '((@first_name|last_name:(%%Heidi%%)) (@first_name:"Heidi" @last_name:"Hegi"))' also, and it too didn't return any record, but i get a record for query, '(@first_name:"Heidi" @last_name:"Hegi")'.

**Comment by nafraf** [comment]
> Hi @nafraf , Thanks for the reply! I tested it with Dialect 2 and it still didn't work. Can you check your Dialect 2 queries again, because you have tested Dialect 2 on same queries. 

Hi @anindyasudhir 
Sorry, my fault, here is the missing test. are we using different conditions for the test?
```sh
127.0.0.1:6379> FLUSHALL
OK
127.0.0.1:6379> HSET doc1 t1 heidi
(integer) 1
127.0.0.1:6379> HSET doc2 t2 heigi
(integer) 1
127.0.0.1:6379> HSET doc3 t1 hegi
(integer) 1
127.0.0.1:6379> HSET doc4 t1 heidi t2 hegi
(integer) 2
127.0.0.1:6379> FT.CREATE idx schema t1 TEXT t2 TEXT
OK
127.0.0.1:6379> module list
1) 1) "name"
   2) "search"
   3) "ver"
   4) (integer) 20612
   5) "path"
   6) "/tmp/redisearch.so"
   7) "args"
   8) (empty array)
127.0.0.1:6379> FT.SEARCH idx '@t1:(%%Heidi%%) | @t2:(%%Heidi%%)' dialect 1 limit 0 0 
1) (integer) 3
127.0.0.1:6379> FT.SEARCH idx '@t1:(%%Heidi%%) | @t2:(%%Heidi%%)' dialect 2 limit 0 0 
1) (integer) 4
```

**Comment by nafraf** [comment]
> I tested it on Query5: '((@first_name|last_name:(%%Heidi%%)) (@first_name:"Heidi" @last_name:"Hegi"))' also, and it too didn't return any record, but i get a record for query, '(@first_name:"Heidi" @last_name:"Hegi")'.

This is the output for Query5, this time I'm using the same field names. I could not reproduce the error.
Are you using any client or are you using redis-cli directly?

```sh
127.0.0.1:6379> FLUSHALL
OK
127.0.0.1:6379> HSET doc1 first_name heidi
(integer) 1
127.0.0.1:6379> HSET doc2 last_name heigi
(integer) 1
127.0.0.1:6379> HSET doc3 first_name hegi
(integer) 1
127.0.0.1:6379> HSET doc4 first_name heidi last_name hegi
(integer) 2
127.0.0.1:6379> FT.CREATE idx schema first_name TEXT last_name TEXT
OK
127.0.0.1:6379> FT.SEARCH idx '((@first_name|last_name:(%%Heidi%%)) (@first_name:"Heidi" @last_name:"Hegi"))' dialect 1 limit 0 0
1) (integer) 1
127.0.0.1:6379> FT.SEARCH idx '(@first_name:"Heidi" @last_name:"Hegi")' dialect 1 limit 0 0
1) (integer) 1
127.0.0.1:6379> FT.SEARCH idx '((@first_name|last_name:(%%Heidi%%)) (@first_name:"Heidi" @last_name:"Hegi"))' dialect 2 limit 0 0
1) (integer) 1
127.0.0.1:6379> FT.SEARCH idx '(@first_name:"Heidi" @last_name:"Hegi")' dialect 2 limit 0 0
1) (integer) 1
```

**Comment by anindyasudhir** [comment]
Hi @nafraf, 
I tested it again and realised that it does work with indexes where the number of documents is not very high, but fails on the indexes with say 10 million documents. I created two indexes with same Schema but I set the prefix in such a way that the smaller index has a subset of all the data while the larger index has all the data. Example:
If hash data present is doc1, doc2, doc11, doc12, doc3 etc
Index1: FT.CREATE idx ON HASH PREFIX 1 doc SCHEMA first_name TEXT last_name TEXT
Index2: FT.CREATE idx2 ON HASH PREFIX 1 doc1 SCHEMA first_name TEXT last_name TEXT

Works with index2 but not with index1.
